format.swift

```diff
@@ -0,0 +1,548 @@
+#!/usr/bin/env swift
+
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+#if canImport(Darwin)
+import Darwin.C
+#elseif canImport(Glibc)
+import Glibc
+#endif
+
+// MARK: - Logging
+
+// It’s a little unfortunate that we need to duplicate all these functions from
+// swift-syntax-dev-utils. But executing format.swift as a script is significantly
+// faster than running it from a package using `swift run` (~1s vs ~15s), which
+// makes a big difference when running the script multiple times.
+
+func logProcessCommand(executableURL: URL?, arguments: [String]?) {
+  var message = ""
+
+  if let executableURL = executableURL {
+    message += executableURL.path
+  }
+
+  if let arguments = arguments {
+    message += " \(arguments.joined(separator: " "))"
+  }
+
+  print(message)
+}
+
+// MARK: - Paths
+
+enum Paths {
+  static var gitExec: URL {
+    get throws {
+      try lookupExecutable(for: "git")
+    }
+  }
+
+  static var swiftExec: URL {
+    get throws {
+      try lookupExecutable(for: "swift")
+    }
+  }
```

Comment:
Could compute these up front and just fatal error rather than throw? Same with most of these computed properties really.

Reply:
I don’t really see any benefit in doing that. For example, there’s no reason to stop script execution if `diff` cannot be found (for verification that code-gen generated the right files) but all you want to do is format swift-syntax using swift-format.

Reply:
Just seems weird to compute them every time 🤷 Reasonable point with `diff` though. I don't really mind in any case, it's just for formatting.